docs/3.0rc/resources/upgrade-prefect-3.mdx

```diff
@@ -0,0 +1,223 @@
+	    return state
+    ```
+  </Card>
+</CardGroup>
+
+In Prefect 3, `PrefectFutures` resulting from submitted task runs are not automatically awaited/resolved before exiting a flow, unless they’re a dependency of other flows or tasks, so *you must wait for a task’s `PrefectFuture` resolution before the end of a flow*:
+
+<CardGroup cols={2}>
+  <Card title="Prefect 2">
+    ```python
+    @task
+    def my_task():
+	    ...
+	
+	
+    @flow
+    def my_flow():
+	    # The flow will not exit until these 
+	    # futures are resolved
+	    [my_task.submit() for i in range(10)]
+    ```
+  </Card>
+  <Card title="Prefect 3">
+    ```python
+    @task
+    def my_task():
+	    ...
+	
+	
+    @flow
+    def my_flow():
+	    futures = [my_task.submit() for i in range(10)]
+	    # You must wait for futures before exiting flow.
+	    futures.wait()
+    ```
+  </Card>
+</CardGroup>
+
+If you do not wait for the futures, the tasks may fail with a 'Flow must be running to start a task' error. 
+The one exception to this case is that `PrefectFuture`s returned from a flow will resolve to states before completing the flow.
+
+## Result persistence and caching behavior
+
+Prefect 3 supports transactional semantics - tasks runs can be grouped into larger transactions, enabling more resilient, idempotent workflows. 
+To support this, Prefect 3 has more eager, intelligent caching defaults than Prefect 2.
+
+Whenever a task runs, Prefect 3 first computes its cache key, then uses that cache key to evaluate whether that same task has run with the same inputs and as part of the same flow before. 
+If it has, the cached result from the previous run is used.
+If it hasn't, the task runs. 
+This behavior can be configured through the new global `cache_policy` setting, or through the `cache_key_fn` on the task, just as in Prefect 2.
+
+Results and cache keys are not stored with your Prefect Cloud account or Prefect server instance. 
+They are persisted entirely in your local result location. 
+Result persistence, and the caching behavior it enables, can be disabled through the `persist_result=False` flag.
+
+See the [caching documentation](/3.0rc/develop/write-tasks#caching) for more information.
+
+## Notifications are now Automations
+
+The notifications feature in Prefect 2 has been expanded to enable many kinds of “if this, then that” logic that can be defined and executed by a Prefect server instance or Prefect Cloud.
+See the [Automations documentation](/3.0rc/automate/events/automations-triggers) for more information.
+
+## Integrations must be upgraded
+
+Integrations compatible with Prefect 2 generally cannot be used with Prefect 3 and vice versa. 
+You must update all of your integrations packages in conjunction with upgrading from Prefect 2 to Prefect 3. 
+We reccomend using the extras on the `prefect` package to ensure that you install the version (e.g. `pip install prefect[aws]`) of the integration that corresponds to your Prefect version.
```

Comment:
```suggestion
We recommend specifying extras when you install the `prefect` package to ensure that you install the integration package version that corresponds to your Prefect version. For example, use `pip install prefect[aws] --pre` to install  Prefect 3 and a compatible prefect-aws version.
```

Reply:
I know there's a PR to add update functionality here, but as of now `pip install prefect[aws]` doesn't work yet and 
`pip install prefect[aws] --pre` should work and give you a Prefect 3 version soon. Without the `--pre` it won't install Prefect 3 on June 25. Also, we might want to add -U for folks upgrading.